### ISSUE TITLE
`relatedItems` is erroring when `publicationYear` is not supplied

### ISSUE THREAD

**Issue by digitaldogsbody** [datacite/lupo#782]
Omitting `publicationYear` from any item in the `relatedItems` array causes the REST API to return an error, but only when `event` is `publish` or `register`:

```json
{
  "errors": [
    {
      "source": "metadata",
      "uid": "10.0330/2rye-vr52",
      "title": "Is invalid"
    },
    {
      "source": "metadata",
      "uid": "10.0330/2rye-vr52",
      "title": "Is invalid"
    }
  ]
}
```

This property is not required by the Metadata Schema 4.4, and should not cause an error if it's missing.

Reproducible with this payload:
```json
{
  "data": {
    "type": "dois",
    "attributes": {
      "prefix": "10.0330",
      "event": "publish",
      "creators": [
        {
          "name": "Bennett, Mike",
          "givenName": "Mike",
          "familyName": "Bennett",
          "affiliation": [
            "DataCite"
          ]
        }
      ],
      "titles": [
        {
          "title": "relatedItems -> publicationYear error test"
        }
      ],
      "publisher": "Datacite",
      "publicationYear": 2022,
      "types": {
        "resourceTypeGeneral": "Dataset"
      },
      "url": "https://example.com",
      "relatedItems": [
        {
          "titles": [
            {
              "title": "Example Related Item"
            }
          ],
          "relatedItemIdentifier": {
            "relatedItemIdentifier": "https://example.org",
            "relatedItemIdentifierType": "URL"
          },
          "relatedItemType": "Dataset",
          "relationType": "IsPartOf",
        }
      ]
    }
  }
}
```

Removing `"event": "publish"` (i.e forcing the DOI to be a draft only) or leaving it and adding `"publicationYear": "2022"` to the `relatedItem` object allows the DOI to submit successfully.

**Comment by digitaldogsbody** [comment]
This looks like a bug with bolognese / the schema as the error message refers to the blank string `''` generated when this element is `null` not matching the RegEx for YearType of `[\d]{4}`, despite the fact that `publicationYear` for related items in the schema is minoccurs 0

**Comment by digitaldogsbody** [comment]
Further testing seems to indicate the problem is here: https://github.com/datacite/bolognese/blob/884a267f09f6d92e24ec922fc59d114b71d2e47a/lib/bolognese/datacite_utils.rb#L256

Other parts of the XML construction use the `xml.something(xyz) if xyz.present?` pattern, which prevents the element being set if it is blank. Since the Related Items setup in bolognese does not do this, attributes get included as self-terminating blank elements (i.e `<publicationYear />`) and this means the schema validation fails.

**Comment by digitaldogsbody** [comment]
This also means that related items created via the API (as opposed to via XML) end up with a lot of these for each related item:

```xml
<relatedItem relatedItemType="Journal" relationType="IsPublishedIn">
    <relatedItemIdentifier relatedItemIdentifierType="DOI">10.1016/j.physletb.2017.11.044</relatedItemIdentifier>
    <creators/>
    <titles>
        <title>Physics letters / B</title>
    </titles>
    <publicationYear>2018</publicationYear>
    <volume>776</volume>
    <issue/>
    <number/>
    <firstPage/>
    <lastPage/>
    <publisher/>
    <edition/>
    <contributors/>
</relatedItem>
```

**Comment by digitaldogsbody** [comment]
Fixed in bolognese 1.9.18
